Fix the segfault error which occurs after Smile provided an update for sorting issue in Elastic Suite

The culprit: https://github.com/Smile-SA/elasticsuite/commit/47a0ad66407d820cec4603b6a94af53003f26a46

diff --git a/src/module-elasticsuite-catalog/Model/ResourceModel/Product/Fulltext/Collection.php b/src/module-elasticsuite-catalog/Model/ResourceModel/Product/Fulltext/Collection.php
index 0d50425..4830680 100644
--- a/src/module-elasticsuite-catalog/Model/ResourceModel/Product/Fulltext/Collection.php
+++ b/src/module-elasticsuite-catalog/Model/ResourceModel/Product/Fulltext/Collection.php
@@ -431,9 +431,6 @@ class Collection extends \Magento\Catalog\Model\ResourceModel\Product\Collection
         }
 
         $this->getSelect()->where('e.entity_id IN (?)', ['in' => $docIds]);
-        $orderList = join(',', $docIds);
-        $this->getSelect()->reset(\Magento\Framework\DB\Select::ORDER);
-        $this->getSelect()->order("FIELD(e.entity_id,$orderList)");
 
         $this->originalPageSize = $this->_pageSize;
         $this->_pageSize = false;
